freqtrade/exchange/binance.py

```diff
@@ -0,0 +1,549 @@
+                order['pair'] = self._symbol_to_pair(o['symbol'])
+                order['opened'] = datetime.datetime.fromtimestamp(
+                                      int(o['time'])/1000).isoformat()
+                order['closed'] = datetime.datetime.fromtimestamp(
+                                      int(o['time'])/1000).isoformat()\
+                    if o['status'] == 'FILLED' else None
+                order['rate'] = float(o['price'])
+                order['amount'] = float(o['origQty'])
+                order['remaining'] = int(float(o['origQty']) - float(o['executedQty']))
+
+        return order
+
+    def cancel_order(self, order_id: str, pair: str) -> None:
+
+        symbol = self._pair_to_symbol(pair)
+
+        api_try = True
+        tries = 0
+        max_tries = 1
+
+        while api_try and tries < max_tries:
+            try:
+                tries = tries + 1
+                data = _API.cancel_order(symbol=symbol, orderId=order_id)
+            except Exception as e:
+                h = Binance._handle_exception(e)
+                if h['fatal']:
+                    raise OperationalException('{message} params=({order_id}, {pair})'.format(
+                        message=str(e),
+                        order_id=order_id),
+                        pair=pair)
+                api_try = h['retry']
+                max_tries = h['retry_max']
+                sleep(0.1)
+
+        return data
+
+    def get_pair_detail_url(self, pair: str) -> str:
+        symbol = self._pair_to_symbol(pair, '_')
+        return 'https://www.binance.com/indexSpa.html#/trade/index?symbol={}'.format(symbol)
+
+    def get_markets(self) -> List[str]:
+        try:
+            data = _API.get_all_tickers()
+        except Exception as e:
+            h = Binance._handle_exception(e)
+            if h['fatal']:
+                raise OperationalException('{message}'.format(message=str(e)))
+
+        markets = []
+
+        stake = _EXCHANGE_CONF['stake_currency']
+
+        for t in data:
+            symbol = t['symbol']
+            symbol_stake_currency = symbol[-len(stake):]
+
+            if symbol_stake_currency == stake:
+                pair = self._symbol_to_pair(symbol)
+                markets.append(pair)
+
+        return markets
+
+    def get_market_summaries(self) -> List[Dict]:
+
+        try:
+            data = _API.get_ticker()
+        except Exception as e:
+            h = Binance._handle_exception(e)
+            if h['fatal']:
+                raise OperationalException('{message}'.format(message=str(e)))
+
+        market_summaries = []
+
+        for t in data:
+            market = {}
+
+            # Looks like this one is only one actually used
+            market['MarketName'] = self._symbol_to_pair(t['symbol'])
+
+            market['High'] = t['highPrice']
+            market['Low'] = t['lowPrice']
+            market['Volume'] = t['volume']
```

Comment:
I think it is better to use "quoteVolume", because volume in target currency can be distorted by small price currencies.